client/modules/IDE/components/PreviewFrame.jsx

```diff
@@ -33,12 +33,32 @@ class PreviewFrame extends React.Component {
       messageEvent.data.forEach((message) => {
         const args = message.arguments;
         Object.keys(args).forEach((key) => {
-          if (args[key].includes('Exiting potential infinite loop')) {
+          if (typeof args[key] === 'string' && args[key].includes('Exiting potential infinite loop')) {
             this.props.stopSketch();
             this.props.expandConsole();
           }
         });
       });
+      for (let i = 0; i < messageEvent.data.length;) {
```

Comment:
i can see what's going on here but this is a pretty dense block of code here. is there any way to clean this up a bit? break it out into its own separate function or functions? or maybe this could be done in the `messageEvent.data.forEach` loop above? You can use `Array.prototype.every` instead of `Array.prototype.forEach` break out of the loop early.

Reply:
Yes, the code here is a bit redundant，l will try to clean it up